insert into database for artist albums

abaddonpuff opened the pull request:
Tried to populate the second database with all albums. But currently getting an error

"ValueError: Field 'id' expected a number but got <artist>"

The idea is, if there are no albums, then its a new artist and requires to populate all albums, if the number of albums recorded is less than the current, then update with the missing albums. And eventually having that process running as a worker.

So modified the model to record the number of albums for a certain artist. and tried to insert all the albums per new artist.

abaddonpuff committed:
insert into database for artist albums

bbelderbos commented on webCommander/centralHub/models.py:
```diff
@@ -34,6 +34,7 @@ class SpotifyArtistInfo(models.Model):
     spotify_artist = models.CharField(max_length = 100, unique=True)
     spotify_image = models.URLField(blank=True)
     spotify_popularity = models.PositiveIntegerField(blank=True)
+    spotify_artist_albums_registered = models.PositiveIntegerField(blank=True, editable=True, default=0)
```
are these needed? `blank=True, editable=True`
no warning here?

bbelderbos commented on webCommander/centralHub/spotify/spotifyAPI.py:
```diff
@@ -52,9 +52,10 @@ def get_spotify_artist(artist_name):
+    print(artist_id)
 
-    url = SPOTIFY_BASE_URL + SPOTIFY_ARTISTS + artist_id
+    url = SPOTIFY_BASE_URL + SPOTIFY_ARTISTS + artist_id[3]
```
is artist_id a list or tuple, why 3?

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -80,9 +80,33 @@ def artist_details(request, spotify_artist):
+    print(f'Amount of albums registered: {artist.spotify_artist_albums_registered}')
     albums = SpotifyAlbumTracking.objects.select_related("spotify_user").filter(spotify_user=artist)
 
+    #Is there any way to test what will actually happen on the database before doing so? :/
```
how do you mean? 

at runtime or by adding code?

at runtime you can add a breakpoint() here to first investigate

but I guess you mean in terms of adding code here?

let's walk through this scenario (or the scenarios) together later ...

bbelderbos committed:
use named tuple for return, use get or create to insert them once

bbelderbos committed:
add type hints

bbelderbos committed:
fix
ValueError: Field 'id' expected a number but got <string>
in
created, album = SpotifyAlbumTracking.objects.get_or_create( ...

bbelderbos committed:
better distinguish variable names

bbelderbos committed:
oops, flipped around return variables

bbelderbos committed:
fixed attribute error, also made the album name column a bit wider in
case we get longer album names back (did not double-check spotify api
though if there is a limit there ...)

bbelderbos committed:
more useful str on model

bbelderbos committed:
handle exception

bbelderbos committed:
fix placement of message, need to be out / at the end of the loop

bbelderbos commented:
Fixed some things here: https://github.com/abaddonpuff/commandCenter/pull/41
Please check and merge all in and start a new branch ...

abaddonpuff committed:
Merge pull request #41 from abaddonpuff/get_albums

use named tuple for return, use get or create to insert them once
